api/gateway/basecommerce/tests/BaseCommerceTest.cfc

```diff
@@ -0,0 +1,311 @@
+		assertTrue(credit.getTransactionId() > 0, 'Invalid transaction id returned: #credit.getTransactionId()#');
+		assertTrue(credit.getParsedResult().Status == 'CREATED', 'Invalid transaction status: #credit.getParsedResult().Status#');
+		assertTrue(credit.getParsedResult().type == 'CREDIT', 'Incorrect transaction type, should be "CREDIT", is: "#credit.getParsedResult().type#"');
+	}
+
+	public void function testDebitAccount() {
+		//Create account
+		local.argumentCollection = structNew();
+		local.argumentCollection.account = createAccount();
+		local.argumentCollection.options.name = 'Test Account';
+		offlineInjector(gw, this, 'mockCreateAccountOk', 'accountData');
+		accountToken = gw.store(argumentCollection = local.argumentCollection);
+		standardResponseTests(accountToken);
+		assertTrue(accountToken.getParsedResult().Status == 'ACTIVE', 'Invalid status: #accountToken.getParsedResult().Status#');
+		assertTrue(accountToken.getParsedResult().type == 'CHECKING', 'Incorrect bank account type, should be "Checking", is: "#accountToken.getParsedResult().type#"');
+		assertTrue(accountToken.getTokenId() != '', 'Token not returned');
+
+		//Debit account
+		local.options = structNew();
+		local.options.tokenId = accountToken.getTokenId();
+		offlineInjector(gw, this, 'mockDebitAccountOk', 'transactionData');
+		debit = gw.purchase(money = variables.svc.createMoney(400, 'USD'), options = local.options);
+		standardResponseTests(debit);
+		assertTrue(debit.getParsedResult().accountType == 'CHECKING', 'Incorrect bank account type, should be "Checking", is: "#debit.getParsedResult().accountType#"');
+		assertTrue(debit.getParsedResult().amount == 4, 'The credit amount requested and the actual credit given is different, should be 4, is: #debit.getParsedResult().amount#');
+		assertTrue(debit.getTransactionId() > 0, 'Invalid transaction id returned: #debit.getTransactionId()#');
+		assertTrue(debit.getParsedResult().Status == 'CREATED', 'Invalid transaction status: #debit.getParsedResult().Status#');
+		assertTrue(debit.getParsedResult().type == 'DEBIT', 'Incorrect transaction type, should be "CREDIT", is: "#debit.getParsedResult().type#"');
+	}
+
+	public void function testCreateInvalidAccountFails() {
+		local.argumentCollection = structNew();
+		local.accountNumberString = '7';
+		local.argumentCollection.account = createInvalidAccount(local.accountNumberString);
+		local.argumentCollection.options.name = 'Test Invalid Account';
+		offlineInjector(gw, this, 'mockCreateInvalidAccountFails', 'accountData');
+		accountToken = gw.store(argumentCollection = local.argumentCollection);
+		standardErrorResponseTests(accountToken, 400);
+		assertTrue(accountToken.getMessage()[1] == 'Account Number must be at least 5 digits', 'Incorrect error message: "#accountToken.getMessage()[1]#", expected: "Account Number must be at least 5 digits"');
```

Comment:
This is not necessary since the standardErrorResponseTests() does all this as part of its tests. If it makes it out of standardErrorResponseTests() then we can assume we have at least one message in array form in getMessage(), and in this case the first one is the message we're interested in.